frontend/express/app.js

```diff
@@ -1431,6 +1434,62 @@ app.post(countlyConfig.path+'/graphnotes/delete', function (req, res, next) {
     }
 });
 
+app.get(countlyConfig.path+'/render', function(req, res){
+    if (!req.session.uid) {
+        return res.redirect(countlyConfig.path+'/login');
+    }
+    
+    var options = {};
+    var view = req.query.view || "";
+    var route = req.query.route || "";
+    var id = req.query.id || "";
+
+    options.view = view + "#" + route;
+    options.id = id ? "#" + id : "";
+    options.savePath = path.resolve(__dirname, "./public/images/scrn" + id + "_" + Date.now() + ".png");
+
+    ip.getHost(function(err, host){
+        if(err){
+            console.log(err);
+            return res.send(false);
+        }
+
+        options.host = host;
+
+        authorize.save({db:countlyDb, multi:false, owner:req.session.uid, purpose:"LoginAuthToken", callback:function(err, token){
+            if(err){
+                console.log(err); 
+                return res.send(false);
+            }
+    
+            options.token = token;
+            render.renderView(options, function(err, image){
+                if(err){
+                    return res.send(false);
+                }
+
+                return res.send(true);
+            });
+        }});
+    });
+});
+
+app.get(countlyConfig.path+'/login/token/:token', function(req, res){
+    var token = req.params.token;
+    var pathUrl = req.url.replace(countlyConfig.path, "");
+    var urlParts = url.parse(pathUrl, true);
+    var fullPath = urlParts.pathname;
+
+    authorize.verify_return({db:countlyDb, token:token, req_path:fullPath, callback:function(valid){
+        if(!valid){
```

Comment:
Since it is form of authorization, it is a security concern and should be logged to System logs plugin. So basically this place should dispatch events `tokenLoginSuccess` and `tokenLoginFailed` like this (just use other name, not apiKey, but tokenLogin, etc):
https://github.com/Countly/countly-server/blob/master/frontend/express/app.js#L1042

And systemlogs plugin should listen to it and record attempts, like this:
https://github.com/Countly/countly-server/blob/master/plugins/systemlogs/frontend/app.js#L59-L65

And of course it needs to be localized too:
https://github.com/Countly/countly-server/blob/master/plugins/systemlogs/frontend/public/localization/systemlogs.properties#L33-L34